avocado/plugins/resolvers.py

```diff
@@ -195,3 +197,84 @@ def resolve(self, reference):
             return criteria_check
 
         return self._validate_and_load_runnables(reference)
+
+
+class ExecRunnablesRecipeInit(Init):
+    name = "exec-runnables-recipe"
+    description = 'Configuration for resolver plugin "exec-runnables-recipe" plugin'
+
+    def initialize(self):
+        help_msg = (
+            'Whether resolvers (such as "exec-runnables-recipe") should '
+            "execute files given as test references that have executable "
+            "permissions. This is disabled by default due to security "
+            "implications of running executables that may not be trusted."
+        )
+        settings.register_option(
+            section="resolver",
+            key="run_executables",
+            key_type=bool,
+            default=False,
+            help_msg=help_msg,
+        )
+
+
+class ExecRunnablesRecipeResolver(BaseExec, Resolver):
+    name = "exec-runnables-recipe"
+    description = "Test resolver for executables that output JSON runnable recipes"
+    priority = PluginPriority.LOW
+
+    def resolve(self, reference):
+        if not self.config.get("resolver.run_executables"):
+            return ReferenceResolution(
+                reference,
+                ReferenceResolutionResult.NOTFOUND,
+                info=(
+                    "Running executables is not enabled. Refer to "
+                    '"resolver.run_executables" configuration option'
+                ),
+            )
+
+        exec_criteria = self.check_exec(reference)
+        if exec_criteria is not None:
+            return exec_criteria
+
+        try:
+            process = subprocess.Popen(
+                reference,
+                stdin=subprocess.DEVNULL,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+        except (FileNotFoundError, PermissionError) as exc:
+            return ReferenceResolution(
+                reference,
+                ReferenceResolutionResult.NOTFOUND,
+                info=(f'Failure while running running executable "{reference}": {exc}'),
+            )
+
+        content, _ = process.communicate()
+        try:
+            runnables = json.loads(content)
```

Comment:
This whole part from 258 to 280 is duplication of `RunnablesRecipeResolver` code. IMO, it would be better to call the `RunnablesRecipeResolver` or create some Base class for holding the json validation a resolving code.

Reply:
I tried to prevent duplication (the 2nd commit is proof), but I found that trying to not duplicate this code (because of the different resolution informational messages, which I believe are a good thing) turned out to bring more complexity than I thought it was worth saving in LOC.

Also, if you look closely this is not a lot statements, with some statements taking like 6 lines because of formatting.

Let me know if you're OK with the current solution.

Reply:
Sure, I am ok with keeping it as it is.